test/gtest/uct/test_sockaddr.cc

```diff
@@ -1043,6 +1061,98 @@ UCS_TEST_P(test_uct_sockaddr, ep_disconnect_err_codes)
 UCT_INSTANTIATE_SOCKADDR_TEST_CASE(test_uct_sockaddr)
 
 
+class test_uct_sockaddr_listener : public test_uct_sockaddr
+{
+protected:
+    test_uct_sockaddr_listener()
+    {
+        m_close_listener_notify = false;
+        m_server_uct_listener   = nullptr;
+    }
+
+    static void conn_request_in_parallel_with_listener_close_cb(
+            uct_listener_h listener, void *arg,
+            const uct_cm_listener_conn_request_args_t *conn_req_args)
+    {
+        test_uct_sockaddr_listener *self =
+                static_cast<test_uct_sockaddr_listener*>(arg);
+
+        self->server_handle_conn_request(
+                listener, conn_req_args,
+                test_uct_sockaddr::server_connect_cb,
+                test_uct_sockaddr::server_disconnect_cb);
+
+        ASSERT_EQ(nullptr, m_server_uct_listener);
+
+        // Notify listener_close_thread to destroy the listener
+        m_close_listener_notify = true;
+
+        // Wait until uct_listener_destroy is being called from
+        // listener_close_thread and dereference the server's listener to check
+        // whether the object is valid or not
+        while (m_server_uct_listener == nullptr) {
+            std::this_thread::yield();
+        }
+
+        ASSERT_TRUE(m_server_uct_listener.load()->cm != nullptr);
+
+        self->server_conn_req_recv();
+    }
+
+    static void listener_close_thread(test_uct_sockaddr_listener *self)
+    {
+        while (!m_close_listener_notify) {
+            std::this_thread::yield();
+        }
+
+        self->m_server_uct_listener = self->m_server->revoke_listener();
+        uct_listener_destroy(self->m_server_uct_listener);
+    }
+
+protected:
+    static std::atomic<bool>           m_close_listener_notify;
+    static std::atomic<uct_listener_h> m_server_uct_listener;
```

Comment:
why needed?

Reply:
can we avoid adding new class and add the test case simpler?

Reply:
> can we avoid adding new class and add the test case simpler?

done

Reply:
updated code does not add new class but it's not simpler, it adds new members and methods to class where it's not needed , if it's impossible to avoid new members, beter to have them in separate class as well as corresponding methods

Reply:
yes, it is impossible to not have them.
moved to separate class as it was